gslib/tests/test_command.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for command.py."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+
+import random
+import string
+import unittest
+from unittest import mock
+
+from gslib import command
+from gslib import command_argument
+from gslib import exception
+from gslib.tests import testcase
+from gslib.utils import constants
+from gslib.utils import shim_util
+
+
+class FakeCommandWithGcloudStorageMap(command.Command):
+  """Implementation of a fake gsutil command."""
+  command_spec = command.Command.CreateCommandSpec('fake',
+                                                   min_args=1,
+                                                   max_args=constants.NO_MAX,
+                                                   supported_sub_args='rz:',
+                                                   file_url_ok=True)
+  gcloud_storage_map = shim_util.GcloudStorageMap(
+      gcloud_command='objects fake',
+      flag_map={
+          '-r': shim_util.GcloudStorageFlag(gcloud_flag='-x'),
+          '-z': shim_util.GcloudStorageFlag(gcloud_flag='--zip'),
+      })
+  help_spec = command.Command.HelpSpec(
+      help_name='fake_command',
+      help_name_aliases=[],
+      help_type='command_help',
+      help_one_line_summary='Fake one line summary for the command.',
+      help_text='Help text for fake command.',
+      subcommand_help_text={},
+  )
+
+
+class FakeCommandWithSubCommandWithGcloudStorageMap(command.Command):
+  """Implementation of a fake gsutil command."""
+  command_spec = command.Command.CreateCommandSpec(
+      'fake_with_sub',
+      min_args=1,
+      max_args=constants.NO_MAX,
+      supported_sub_args='ay:',
+      file_url_ok=True,
+      argparse_arguments={
+          'set': [
+              command_argument.CommandArgument.
```

Comment:
I'm surprised this doesn't need to be wrapped in parenthesis:
```
(command_argument.CommandArgument.
 MakeZeroOrMoreCloudBucketURLsArgument())
```

Reply:
Yeah, yapf behaves strangely sometimes, and I don't always like it the way it formats things.